pkg/parse/quote.go

```diff
@@ -99,26 +104,33 @@ func rtohex(r rune, w int) []byte {
 func quoteDouble(s string) string {
 	var buf bytes.Buffer
 	buf.WriteByte('"')
-	for _, r := range s {
```

Comment:
It's a bit cleaner to keep the `range` iteration, and you can still access the underlying byte using the index:

```go
for i, r := range s {
  if r == utf8.RuneError{
    // s[i] contains the invalid byte
  }
}
```

Reply:
Yes, but as I mentioned in a prior comment it is better to use the utf8 package decode rune pattern consistently. That makes it easier for readers of the code to reason about the behavior. It also appears to be the current consensus in the Go community as a best practice for code that needs to iterate over strings/byte-slices that might contain invalid UTF-8. I've also seen comments by people like Ian Taylor (i.e., Go core developers) expressing regret about the current `range string` behavior.